apollo-router/src/plugins/file_uploads/rearange_query_plan.rs

```diff
@@ -0,0 +1,244 @@
+use std::cmp;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+use indexmap::IndexSet;
+use itertools::Itertools;
+
+use super::error::FileUploadError;
+use super::MapPerVariable;
+use super::UploadResult;
+use crate::query_planner::DeferredNode;
+use crate::query_planner::FlattenNode;
+use crate::query_planner::PlanNode;
+use crate::services::execution::QueryPlan;
+
+pub(super) fn rearange_query_plan(
```

Comment:
I would be happy to have a little bit of docs here. To know what's the goal of this function